### PR TITLE
feat: support task listeners

### PR DESCRIPTION
**Pull request by misiekhardcore**
related to https://github.com/camunda/camunda-modeler/issues/4590

Contains:

- feat: add `no-duplicate-task-listeners` rule - to prevent duplicates rejected by the engine
- feat: add `no-task-listeners` rule - to prevent usage with unsupported engine version
- feat: add `task-listener` rule - to ensure correct configuration

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ const camundaCloud10Rules = withConfig({
   'no-propagate-all-parent-variables': 'error',
   'no-task-schedule': 'error',
   'no-template': 'error',
+  'no-task-listeners': 'error',
```
Comment (barmac) [inline index.js:24]: Let's keep the alphabetic order.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,12 @@ All notable changes to [bpmnlint-plugin-camunda-compat](https://github.com/camun
 
 ___Note:__ Yet to be released changes appear here._
 
+* `DEPS`: bump `zeebe-bpmn-moddle@1.7.0`
+* `FEAT`: support task listeners ([#181](https://github.com/camunda/bpmnlint-plugin-camunda-compat/pull/181))
+* `FEAT`: add `no-duplicate-task-listeners` rule
```
Comment (barmac) [inline CHANGELOG.md:11]: There was a decision to not reject duplicate listeners, but we can still show a warning. Cf. https://github.com/camunda/camunda-modeler/issues/4590#issuecomment-2446717471

Comment (barmac) [reply CHANGELOG.md:11]: I'd say let's not add this rule to the compat project. We can open a discussion on a different set of rules. Perhaps bpmnlint-plugin-camunda is the right place for the best practices?

**Review by barmac — APPROVED** [review]
✅ with duplicate-listeners removed

File: CHANGELOG.md (hunk 2 above)
Comment (misiekhardcore) [reply CHANGELOG.md:11]: As discussed, I am removing this completely

**Pull request merged**